plugins/jvm/ts/discover.ts

```diff
@@ -17,19 +17,32 @@ namespace JVM {
       (<any>$)($event.currentTarget).parents('.popover').prev().popover('hide');
     };
 
+    function getMoreJvmDetails(agents){
+        for(let key in agents) {
+            const agent=agents[key];
+            if(agent.url && !agent.secured ) {
+                const dedicatedJolokia=createJolokia(agent.url, agent.username, agent.password);
+                agent.startTime=dedicatedJolokia.getAttribute('java.lang:type=Runtime', 'StartTime');
+                if(!$scope.hasName(agent)){//only look for command if agent vm is not known
+                    agent.command=dedicatedJolokia.getAttribute('java.lang:type=Runtime', 'SystemProperties', 'sun.java.command');
+                }
+            }
+        }
+    }
+
     function doConnect(agent) {
       if (!agent.url) {
         Core.notification('warning', 'No URL available to connect to agent');
         return;
       }
-      var options:Core.ConnectToServerOptions = Core.createConnectOptions();
-      options.name = agent.agent_description;
-      var urlObject = Core.parseUrl(agent.url);
+      const options:Core.ConnectToServerOptions = Core.createConnectOptions();
+      options.name = agent.agent_description  || 'discover-' + agent.agent_id;
```

Comment:
Need to use unique name to get it saved to local storage and picked up in new tab